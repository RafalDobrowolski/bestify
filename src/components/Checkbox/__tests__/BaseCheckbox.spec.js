import { mount } from '@vue/test-utils';
import bCheckbox from "../BaseCheckbox";

describe('Checkbox', () => {
  test('Checkbox is rendered correctly', () => {
    const wrapper = mount(bCheckbox);
    expect(wrapper.html()).toMatchSnapshot()
  });

  test('Checkbox is disabled', () => {
    const wrapper = mount(bCheckbox, {
      propsData: {
        disabled: true
      }
    });
    expect(wrapper.find('input').attributes().disabled).toBeTruthy();
  });

  test('Checkbox when value is changed', () => {
    const wrapper = mount(bCheckbox, {
      propsData: {
        value: true
      }
    });

    expect(wrapper.find('input').attributes().value).toBeTruthy()
  });

  test('Should emit an input event',  () => {
    const wrapper = mount(bCheckbox);
    wrapper.find('input').trigger('change');
    expect(wrapper.emitted().input).toBeTruthy()
    expect(wrapper.emitted().input).toHaveLength(1);
  });

});