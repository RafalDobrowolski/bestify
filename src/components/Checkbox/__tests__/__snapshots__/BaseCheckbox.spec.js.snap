// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Checkbox is rendered correctly 1`] = `<label class="checkbox"><input type="checkbox" class="checkbox__input" value="false"> <span>check box</span></label>`;
